plugin-server/src/main/utils.ts

```diff
@@ -34,3 +40,61 @@ export async function runInstrumentedFunction({
         clearTimeout(timeout)
     }
 }
+
+export async function kafkaHealthcheck(
+    kafka: Kafka,
+    statsd?: StatsD,
+    timeoutMs = 20000
+): Promise<[boolean, Error | null]> {
+    let consumer: Consumer | null = null
+    let producer: Producer | null = null
+
+    try {
+        producer = kafka.producer()
+        await producer.connect()
+        await producer.send({
+            topic: 'healthcheck',
+            messages: [
+                {
+                    partition: 0,
+                    value: Buffer.from('healthcheck'),
+                },
+            ],
+        })
+
+        let kafkaConsumerWorking = false
+        consumer = kafka.consumer({
+            groupId: 'healthcheck-group',
+        })
+
+        await consumer.subscribe({ topic: 'healthcheck', fromBeginning: true })
```

Comment:
nope - the whole point here is to test that we can set up a new consumer, subscribe, and consume correctly. It's mostly testing for things like connection errors.

Reply:
ok let me clarify what I meant - say we have successfully connected to kafka, then we have a healthcheck message in that topic, so any later attempt would only verify that the consumer works, no (is the producer failed and didn't throw we wouldn't know because of the fromBeginning here)?

Reply:
No? Consumer and producer are separate entities - we're always producing a new message in this flow here

Reply:
to the same topic though, no? So the previous messages in that topic still exist and consumers and producers are tied to the topic. So if my producer restarts the topic won't be empty is my understanding of the world.

Reply:
Why should it have to be empty?

Reply:
I originally wrote this to consume the same message but that's actually a pattern that does not work here because some other plugin server might have that partition assigned to it. Can't guarantee a given server will get a given message (I thought quite a bit about this). 

And actually I realize this pattern here might have an issue with partitions still. One approach it to always use a new consumer group but that's not a good idea from a Kafka perspective - it'd eventually have to store a lot of data.

Reply:
will rethink a bit

Reply:
My solution here relies on each partition having at least one message, which is not guaranteed ofc.

Also setting partitions to 1 would prevent other consumers from working 🤔

Reply:
changed it so that if the consumer is up and fetching messages we mark it as healthy